# Insurance Data Starter Notebook

Student Name: Samuel Valdes Gutierrez



## Table of Contents


This notebook will be used for 4 practice assignments:
- [📝 Loading Data (Practice)](#loading)
- [📝 Slicing (Practice)](#slicing)
- [📝 Filtering (Practice)](#filtering)


Click on the assignment names above to jump to the corresponding headers on Google Colab.  



___




Note that:
  - The first assignment (Loading Data) must be completed before the other three assignments may be completed.

  - We recommend you do each of the remaining sections as you encounter them on the Learning Platform.


<a name="loading"> </a>
# 📝 Loading Data (Practice)

In [3]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


[Insurance Data](https://drive.google.com/file/d/1-7mcBguuzAV7JWVu2XCpgAO-fO6X2cl0/view?usp=sharing)

1) Open the link and save the shared file to your Drive in the week 2 data folder.

2) Mount your drive

3) Find and copy the file path of the insurance_mod.csv file

In [4]:
# Save the copied file path as a variable
# Surround the path with quotes
file_path = "/content/drive/MyDrive/Coding_Dojo_Data_Science_Excercises/02_Pandas_for_Data_Manipulation/data/insurance_mod.csv"

4) Load in the file as a Pandas dataframe named df_insurance


In [5]:
# Include needed import(s)
import pandas as pd

# Load in the file as a Pandas dataframe named df_insurance
df_insurance = pd.read_csv(file_path)

5) Preview the first 5 rows with .head()



In [6]:
# Preview the first 5 rows with .head()
df_insurance.head()

age     sex     bmi  children  smoker     region  charges
0   19  female  27.900         0       1  southwest  16885.0
1   18    male  33.770         1       0  southeast   1726.0
2   28    male  33.000         3       0  southeast   4449.0
3   33    male  22.705         0       0  northwest  21984.0
4   32    male  28.880         0       0  northwest   3867.0

6) Preview the last 5 rows with .tail()

In [7]:
# Preview the last 5 rows with .tail()
df_insurance.tail()

age     sex    bmi  children  smoker     region  charges
1333   50    male  30.97         3       0  northwest  10601.0
1334   18  female  31.92         0       0  northeast   2206.0
1335   18  female  36.85         0       0  southeast   1630.0
1336   21  female  25.80         0       0  southwest   2008.0
1337   61  female  29.07         0       1  northwest  29141.0

7) Obtain summary information with .info()

In [8]:
# Obtain summary information with .info()
df_insurance.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1338 entries, 0 to 1337
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype  
---  ------    --------------  -----  
 0   age       1338 non-null   int64  
 1   sex       1338 non-null   object 
 2   bmi       1338 non-null   float64
 3   children  1338 non-null   int64  
 4   smoker    1338 non-null   int64  
 5   region    1338 non-null   object 
 6   charges   1338 non-null   float64
dtypes: float64(2), int64(3), object(2)
memory usage: 73.3+ KB


8) Use the output of .info() to answer the following in a text cell below:




a) How many rows are there?

- Answer: 1,338 rows

b) How many columns are there?

- Answer: 7 cooumns

## You have completed the "Loading Data" Practice Assignment.



- 🛑 Stop here.

- 💾 Save and submit this notebook now.

- 📚 Continue working through the lessons.

- ⏎ Return to your copy of this notebook for the next practice assignment.

___


<a name="slicing"> </a>
# 📝 Slicing (Practice)


1) Slice the row for the customer at index 50 as a Series

In [9]:
# Use single brackets so the output is a Pandas Series
df_insurance.loc[50]

age                18
sex            female
bmi            35.625
children            0
smoker              0
region      northeast
charges        2211.0
Name: 50, dtype: object

2) Slice the row for the customer at index 50 as a DataFrame

In [10]:
# Use double brackets so the output is a Pandas Dataframe
df_insurance.loc[[50]]

age     sex     bmi  children  smoker     region  charges
50   18  female  35.625         0       0  northeast   2211.0

3) Find out if the customer at index 20 was a male or female (by slicing index 20 and column "Sex")

In [11]:
# find out if the customer at index 20 was a male or female by slicing index 20 and column "Sex"
df_insurance.loc[50, "sex"]

'female'

4) Slice the entire smoker column using square brackets


In [12]:
# Slice the entire smoker column using square brackets
df_insurance["smoker"]

0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

5) Slice the entire smoker column using .loc


In [13]:
# Slice the entire smoker column using .loc
df_insurance.loc[:, "smoker"]

0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64

6) Slice the entire smoker column using either method, but this time save it as a variable called "smoker"


In [14]:
# Slice the entire smoker column using either method, but this time save it as a variable called "smoker"
smoker = df_insurance.loc[:, "smoker"]
smoker

0       1
1       0
2       0
3       0
4       0
       ..
1333    0
1334    0
1335    0
1336    0
1337    1
Name: smoker, Length: 1338, dtype: int64


- Calculate the sum of the smoker variable to see how many of the customers smoke


In [15]:
# To see how many of the customers smoke, get the sum
smoker.sum()

274


- Calculate the mean of the smoker variable using .mean() to see the ratio of smokers in the dataset


In [16]:
# To get the ratio use .mean()
smoker.mean()

0.20478325859491778

## You have completed the "Slicing Data" Practice Assignment.



- 🛑 Stop here.

- 💾 Save and submit this notebook now.

- 📚 Continue working through the lessons.

- ⏎ Return to your copy of this notebook for the next practice assignment.

___


<a name="filtering"> </a>
# 📝 Filtering (Practice)

1) Create a filter variable called under_30_filter for rows where the age is less than 30



In [17]:
# Create a filter called under_30_filter for the rows where the age is less than 30


In [18]:
# View the under_30_filter to see the boolean index


In [19]:
# Get the sum of the true values for the under_30_filter
# This is the number of customers who are under the age of 30


In [20]:
# # Use the filter to view only rows with customers who are under 30


2) Create a filter for males

In [21]:
# Create a filter for males called male_filter


In [22]:
# View the filter to see the boolean index


In [23]:
# Obtain the sum of True values in the male_filter
# This is the number of customers who are male


In [24]:
# Use the filter to view only rows with customers who are male


3) Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter

In [25]:
# Combine the under_30_filter and the male_filter to create a single filter called under_30_male_filter


In [26]:
# Display the boolean index of the under_30_male_filter


In [27]:
# Get the sum of the under_30_male_filter True values
# This is the number of customers who are both under 30 and male


In [28]:
# Use the filter to view only rows with customers who are both under 30 and male


4) Create a filter for regions that contain south using .str.contains



In [29]:
# Create a filter for regions that contain south using .str.contains


In [30]:
# Use the filter to view only rows with regions containing "south"


## You have completed the "Filtering Data" Practice Assignment.



- 🛑 Stop here.

- 💾 Save and submit this notebook now.

- 📚 Continue working through the lessons.

- ⏎ Return to your copy of this notebook for the next practice assignment.

___
